src/Microsoft.AspNetCore.Mvc.RazorPages/ApplicationModels/PageApplicationModelProviderContext.cs

```diff
@@ -10,9 +11,25 @@ namespace Microsoft.AspNetCore.Mvc.ApplicationModels
     /// </summary>
     public class PageApplicationModelProviderContext
     {
+        public PageApplicationModelProviderContext(PageActionDescriptor descriptor, TypeInfo pageTypeInfo)
+        {
+            ActionDescriptor = descriptor;
+            PageType = pageTypeInfo;
+        }
+
+        /// <summary>
+        /// Gets the <see cref="PageActionDescriptor"/>.
+        /// </summary>
+        public PageActionDescriptor ActionDescriptor { get; }
+
+        /// <summary>
+        /// Gets the page <see cref="TypeInfo"/>.
+        /// </summary>
+        public TypeInfo PageType { get; }
```

Comment:
`PageActionDescriptor` doesn't have a type info (just the paths). We need to pass this to the provider in some format, no?